hatchery/hatchery.go

```diff
@@ -496,7 +496,7 @@ func terminate(w http.ResponseWriter, r *http.Request) {
 		Config.Logger.Printf(userlicerr.Error())
 	}
 	Config.Logger.Printf("Debug: Active gen3 license user maps %v", activeGen3LicenseUsers)
-	if len(*activeGen3LicenseUsers) == 0 {
+	if activeGen3LicenseUsers == nil || len(*activeGen3LicenseUsers) == 0 {
```

Comment:
@george42-ctds said:
> I think the edge case is when /terminate tries to query the table but the table does not exist in the environment. That shows up as errors in the logs.
for that case, should the getActiveGen3LicenseUserMaps return an empty list instead of nil?

Ok, yes i do think `getActiveGen3LicenseUserMaps` should check if the config exists before trying to query, and in any case, return an empty list if the user doesn't have a license.

I think the `getActiveGen3LicenseUserMaps` function should return `gen3LicenseUserMaps []Gen3LicenseUserMap` instead of `gen3LicenseUserMaps *[]Gen3LicenseUserMap`. That should remove a lot of the confusion and the returned type will be enforced, so it will always return a list and the calling functions won't have to check the returned value.

Reply:
Good ideas! 

- output type is updated
- empty list is returned for any DB errors or for no active users
- test case added for function to return empty list (triggered by license not enabled in config).

Reply:
Additional check has been added.